store/kv/atomic.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2022-2023 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kv
+
+import (
+	"context"
+	"math/rand"
+)
+
+const (
+	DefaultNumShards      = 1024
+	AtomicSuffixSeparator = "__A__"
+)
+
+type ShardedAtomics interface {
```

Comment:
As we are introducing this interface, should we move `AtomicAdd`/`AtomicRead` from the kv interface to a `NonShardedAtomics` interface? Or remove them.

Reply:
Not a bad idea. I think how to do it properly.